chore: Add pull request template (#449)
uktrade/platform-tools: .github/pull_request_template.md

Addresses <link to ticket>.

Please add any relevant context for you pull request here, or delete this if none needed.

---
## Checklist:

> - [ ] Ticket reference included in title (unless it's a quick out of ticket thing)

Reviewer: ```suggestion
- [ ] Title: scope included as per [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/)
- [ ] Title: ticket reference included (unless it's a quick out of ticket thing)
```

> - [ ] Link to ticket included in description (unless it's a quick out of ticket thing)

Reviewer: ```suggestion
- [ ] Description: link to ticket included (unless it's a quick out of ticket thing)
```

- [ ] Includes tests (or an explanation for why it doesn't)
- [ ] If the work includes user interface changes, before and after screenshots included in description
- [ ] Includes any applicable changes to the documentation in this code base
- [ ] Includes link(s) to any applicable changes to the documentation in the [DBT Platform Documentation](https://platform.readme.trade.gov.uk/) (can be to a pull request)

> - [ ] I have manually run the commented out sections of the [pull request regression tests](https://github.com/uktrade/platform-tools/blob/main/regression_tests/pull_request_tests.sh) (unless it's not a code change)

Reviewer: ```suggestion
- [ ] Manually run the commented out sections of the [pull request regression tests](https://github.com/uktrade/platform-tools/blob/main/regression_tests/pull_request_tests.sh) (unless it's not a code change)
```
